### PR TITLE
Admin Users Re-work

### PR DESCRIPTION
**Pull request by Michael-Vol**
**What kind of change does this PR introduce?** 
This PR introduces new routes for admin users:
- Get all admin users
- Change current user password (current password required)
- Delete another user password

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other (please describe)

**Does this PR introduce a breaking change?** 

- [x] Yes
- [ ] No

Create admin URI updated in conformance with resource-based convention (POST: `/admin/create` -> `/admin/admins`)
@KostasFeg ^

**The PR fulfills these requirements:**

- [x] It's submitted to the `main` branch

### CODE REVIEW

File: libraries/grpc-sdk/src/interfaces/Indexable.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export type Indexable = { [key: string]: any };
+export type Params = Indexable;
+export type Document = Indexable;
+export type Route = Indexable;
+export type Headers = Indexable;
+export type Context = Indexable;
+export type Query = Indexable;
```
Comment (kon14) [inline libraries/grpc-sdk/src/interfaces/Indexable.ts:7]: Export * from `src/interfaces/index.ts` (and import Query directly from index.ts in ConduitActiveSchema)
Add new line at EOF.

File: packages/admin/src/routes/DeleteAdminUser.route.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  },
+   new ConduitRouteReturnDefinition('DeleteAdminUser', {
+    message: ConduitString.Required,
+  }),
+  async (params: ConduitRouteParameters) => {
+    const { id } = params.params!;
+    if(isNil(id)) {
+      throw new ConduitError('INVALID_ARGUMENTS', 400, 'Id must be provided');
+    }
+    let admin = await Admin.getInstance().findOne({ _id: id });
```
Comment (kkopanidis) [inline packages/admin/src/routes/DeleteAdminUser.route.ts:22]: make sure that an admin cannot delete themselves

**Review by kkopanidis — CHANGES_REQUESTED** [review]

File: packages/router/src/interfaces/Indexable.ts [hunk 3]
```diff
@@ -0,0 +1,2 @@
+export type Indexable = { [key: string]: any };
+export type Params = Indexable;
```
Comment (kon14) [inline packages/router/src/interfaces/Indexable.ts:2]: Why not reuse this from `grpc-sdk`?
(If there's an actual reason for this, export types from `../index.ts` and add new line at EOF.

File: packages/admin/src/routes/index.ts [hunk 4]
```diff
@@ -1,3 +1,6 @@
 export * from './Login.route';
 export * from './GetModules.route';
 export * from './CreateAdmin.route';
+export * from './GetAdminUsers.route';
+export * from './DeleteAdminUser.route';
+export * from './ChangePasswordRoute';
```
Comment (kon14) [inline packages/admin/src/routes/index.ts:6]: New line at EOF.

File: packages/admin/src/routes/DeleteAdminUser.route.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    if(isNil(admin)) {
+      throw new ConduitError('NOT_FOUND', 404, 'Admin not found');
+    }
+    admin = await Admin.getInstance().deleteOne({ _id: id });
+    return { result: {message : 'Admin deleted.' } };
+  }) 
+}
```
Comment (kon14) [inline packages/admin/src/routes/DeleteAdminUser.route.ts:29]: New line at EOF.

File: packages/admin/src/routes/GetAdminUsers.route.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import { ConduitRoute, ConduitRouteReturnDefinition } from '@conduitplatform/commons';
+import { ConduitRouteActions } from '@conduitplatform/grpc-sdk';
+import { Admin } from '../models';
+
+export function getAdminUsersRoute() {
+  return new ConduitRoute(
+    {
+      path: '/admins',
```
Comment (kon14) [inline packages/admin/src/routes/GetAdminUsers.route.ts:8]: Should we also update create admin path as `POST @ /admin/admins` while at it (breaking change).

File: packages/admin/src/middleware/Auth.middleware.ts [hunk 7]
```diff
@@ -47,7 +47,7 @@ export function getAuthMiddleware(grpcSdk: ConduitGrpcSdk, conduit: ConduitCommo
         if (isNil(admin)) {
           return res.status(401).json({ error: 'No such user exists' });
         }
-        (req as any).admin = admin;
+        (req as any).conduit.admin = admin;
```
Comment (kon14) [inline packages/admin/src/middleware/Auth.middleware.ts:50]: Don't we have a generic request type at this point?

**Review by kon14 — CHANGES_REQUESTED** [review]

**Comment by kon14** [comment]
@KostasFeg tagging you here in case OP edit does not proc a notification.
Create admin URI updated in conformance with resource-based convention (POST: /admin/create -> /admin/admins)

**Review by kkopanidis — APPROVED** [review]

**Review by kon14 — APPROVED** [review]

**Pull request merged**